sys/parsoid.js

```diff
@@ -344,7 +392,9 @@ class ParsoidService {
                         body: res.body[format].body
                     };
                     resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-                    return this.saveParsoidResult(hyper, req, format, tid, res)
+                    return this.saveParsoidResultToLatest(hyper, req, tid, res)
+                    .catch({ status: 412 }, () =>
```

Comment:
> The problem with this idea is that on this level we do not know the value of "12345" It's only know inside the parsoid_bucket...

The number we need to know here is the revision we are about to store, which I believe we know.